src/frontend/src/components/SearchBar/SearchBar.jsx

```diff
@@ -92,15 +92,18 @@ function CustomizedInputBase(props) {
 
   const onFilterChange = (event) => {
     onFilterChangeHandler(event.target.value);
+    navigate(`/search?text=${searchText}&filter=${event.target.value}`);
```

Comment:
Let's write a function above these that does a few things:

```js
const prepareParamsAndNavigate = (text, filter) =>
  navigate(`/search?text=${encodeURIComponent(text)}&filter=${encodeURIComponent(filter)}`);
```

Now in all these cases where you call `navigate` you can pass the args to `prepareParamsAndNavigate(text, filter)` and have it fix them up for URL inclusion and navigate.

Reply:
Good suggestion. Need to improve coding skill. lol